### PR TITLE
#18: Base styles and theme

### LINKED ISSUE #18: Setup Base Styles and Theme
- CSS reset
- Theme variable
- Basic utility classes
- Fonts

### PR DESCRIPTION
**Pull request by rpeterman-gp**
Closes #18

- Sets up base styles and theme for app and embed page.
- Adds ability for components to define their own theming that can be customized by other components that use them.
- Update icon svg's to use svg's from Google Fonts > Material Icons. (They have more consistent spacing than the svg's from Figma.)
- Adds hover and focus styling to icon buttons

## To Review

- [ ] Checkout Branch.
- [ ] Run `nvm use`.
- [ ] Run `yarn`.
- [ ] Run `yarn dev`.
- [ ] Go to http://localhost:3000/e?uf=http://feed.songexploder.net/SongExploder.

> ...then...

- [x] Ensure player theming is correct.
- [x] Note that the play button has customized its use of the icon button component theme to have its buttons be brighter and larger.
- [x] Check out the icon buttons' hover and focus states.
- [x] Go to http://localhost:3001/e?uf=http://feed.songexploder.net/SongExploder&ca=1.
- [x] Ensure theming is correct when cover art is shown.
- [x] Go to http://localhost:3001/e?uf=http://feed.songexploder.net/SongExploder&sp=10
- [x] Ensure placeholder playlist theme is correct (further theming will come in the playlist component ticket.)
- [x] Code review `components/IconButton` to see how it uses the ThemeVars component to define its own theming custom variables.
- [ ] Code review `styles` scss files to see what theming sass variables are defined.

### CODE REVIEW

**Comment by brandonhundt** [comment]
Thanks @rpeterman-gp a few thoughts:

- I'm not sure the play/button should be in a circle style like the other buttons. Maybe its something to address once the skip buttons are added, but the intent in the design is that the play button be larger, a different style than the other buttons in the player.
- For a11y, buttons will need to be labelled (`aria-label="play"` `aria-label="pause"`
- Have you pulled in the breakpoint styles in #29 to this branch? If not, I'll ignore this assuming it will get merged in.

**Comment by brandonhundt** [comment]
In the designs I did signal an opacity change. I'm okay with an outline for keyboards and what you came up with in terms of hover and focus states. My concern is just more spacing and the play button being in the circle style. Its a little tough for me to weigh how this will appear without the skip buttons in place. 

In #13, I thought I remembered the thumbnail and text size getting smaller at the mobile breakpoint - thats intended. I'm not seeing that here.

**Comment by rpeterman-gp** [comment]
I am seeing the thumbnail getting smaller and the title font size scaling down when at < 500px viewport width.

File: components/Player/PlayButton/PlayButton.tsx [hunk 1]
```diff
@@ -21,9 +23,14 @@ const PlayButton: React.FC<IPlayButtonProps> = ({ ...props }) => {
   };
 
   return (
-    <button {...props} type="button" onClick={handleClick}>
+    <IconButton
+      className={styles.root}
+      {...props}
+      type="button"
```
Comment (brandonhundt) [inline components/Player/PlayButton/PlayButton.tsx:29]: I suspect this is where the aria-label would appear...

**Review by brandonhundt — APPROVED** [review]
@rpeterman-gp :shipit: I'm looking forward to seeing the play button alongside the skip buttons in a future PR

**Pull request merged**